### PR TITLE
New transform named exp_safe_html

### PR DESCRIPTION
**Pull request by prakharjoshi**
This is a separate pull request just for adding new transform.

### CODE REVIEW

**Comment by prakharjoshi** [comment]
@tisto  @lentinj . Yayaya!! travis works again and its happy now. Thanks @tisto  for helping me out with the travis issue. I just fetch the upstream master to the master of fork and created new branch just for the new transform . Test cases are passing and zope instance is working fine. Have a look :)

File: src/experimental/safe_html_transform/configure.zcml [hunk 1]
```diff
@@ -33,4 +33,9 @@
         handler=".setuphandlers.post_install">
     </genericsetup:importStep>
 
+  <include package="plone.app.registry" />
+  <include package="Products.MimetypesRegistry" />
+  <include package="Products.PortalTransforms" />
+  <include package="Products.PortalTransforms" />
```
Comment (lentinj) [inline src/experimental/safe_html_transform/configure.zcml]: These shouldn't be necessary, as Plone will include the necessary ZCML for you in all of these cases. You definitely don't need to include `Products.PortalTransforms` twice.

File: src/experimental/safe_html_transform/testing.py [hunk 2]
```diff
@@ -23,6 +25,8 @@ def setUpZope(self, app, configurationContext):
             experimental.safe_html_transform,
             context=configurationContext
         )
+        logger.root.setLevel(logging.DEBUG)
+        logger.root.addHandler(logging.StreamHandler(sys.stdout))
```
Comment (lentinj) [inline src/experimental/safe_html_transform/testing.py]: Don't leave logging to stdout on all the time. Either remove this or make it conditional somehow.

File: src/experimental/safe_html_transform/tests/test_safe_html.py [hunk 3]
```diff
@@ -1,205 +0,0 @@
...
-
-    def test_html_full_tag(self):
-        html = "<html></html>"
-        data = datastream(self.transform.name())
-        self.assertEqual(self.transform.convert(html, data), "")
-
-    def test_html_without_ptag(self):
-        html = "<p>Keep me"
-        data = datastream(self.transform.name())
-        self.assertEqual(self.transform.convert(html, data), "<p>Keep me</p>")
```
Comment (lentinj) [inline src/experimental/safe_html_transform/tests/test_safe_html.py:205]: Why are you throwing away the tests?

File: src/experimental/safe_html_transform/transforms/exp_safe_html.py [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+                p = re.compile(r'<.?html?.>')
+                safe_html2 = p.sub('', safe_html2)
+            safe_html2 = safe_html2.replace("h3", "p")
+            safe_html2 = safe_html2.replace("h4", "p")
+            safe_html2 = safe_html2.replace("h5", "p")
+            safe_html2 = safe_html2.replace("h6", "p")
+            safe_html2 = safe_html2.replace("div", "p")
+            data.setData(safe_html2)
+
+        return data._data
```
Comment (lentinj) [inline src/experimental/safe_html_transform/transforms/exp_safe_html.py]: As I've mentioned before. This doesn't return like a compliant transform should.

File: src/experimental/safe_html_transform/configure.zcml (hunk 1 above)
Comment (prakharjoshi) [inline src/experimental/safe_html_transform/configure.zcml]: So I should remove all of them as these are  dependencies in metadata.py for default profile ? And Yes PortalTransform was included twice, Will correct that.

File: src/experimental/safe_html_transform/testing.py (hunk 2 above)
Comment (prakharjoshi) [inline src/experimental/safe_html_transform/testing.py]: Sure Will correct that :)

File: src/experimental/safe_html_transform/configure.zcml (hunk 1 above)
Comment (lentinj) [inline src/experimental/safe_html_transform/configure.zcml]: There are several entirely different dependency mechanisms involved here, don't confuse them.
- Package dependencies in `setup.py`, that cause python code to be available
- ZCML dependencies in `configure.zcml`, that cause that package's ZCML code to be loaded when yours is
- GenericSetup dependencies in `metadata.xml`, that cause that package's GenericSetup profile to be installed whenever yours is installed.

Search for each on docs.plone.org to find out more.

However, the short answer is use the tests. If they still pass, then they weren't necessary.

**Comment by gforcada** [comment]
@prakharjoshi this branch needs to be rebased as the controlpanel branch got merged, some examples: https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase

**Comment by prakharjoshi** [comment]
Closing this one as correct one is #31.

**Comment by prakharjoshi** [comment]
Done. :)

**Pull request closed without merging**